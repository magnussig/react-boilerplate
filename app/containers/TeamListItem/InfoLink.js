import styled from 'styled-components';
import { Link } from 'react-router-dom';

const InfoLink = styled(Link)`
  height: 100%;
  color: black;
  display: flex;
  align-items: center;
  width: 100%;
`;

export default InfoLink;
