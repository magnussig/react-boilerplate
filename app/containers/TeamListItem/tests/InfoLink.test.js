import React from 'react';
import { mount } from 'enzyme';
import { enzymeFind } from 'styled-components/test-utils';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import 'jest-styled-components';

import InfoLink from '../InfoLink';

describe('<InfoLink />', () => {
  it('should match the snapshot', () => {
    const renderedComponent = renderer.create(
      <Router>
        <InfoLink to="/" />
      </Router>
    ).toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });

  /*it('should have a className attribute', () => {
    const wrapper = mount(
      <Router>
        <InfoLink to="/" />
      </Router>);
    const renderedComponent = enzymeFind(wrapper, InfoLink);
    expect(renderedComponent.prop('className')).toBeDefined();
  });

  it('should adopt a valid attribute', () => {
    const id = 'test';
    const wrapper = mount(
      <Router>
        <InfoLink id={id} to="/" />
      </Router>);
    const renderedComponent = enzymeFind(wrapper, InfoLink);
    expect(renderedComponent.prop('id')).toEqual(id);
  });

  it('should not adopt an invalid attribute', () => {
    const wrapper = mount(
      <Router>
        <InfoLink attribute="test" to="/" />
      </Router>);
    const renderedComponent = enzymeFind(wrapper, InfoLink);
    expect(renderedComponent.prop('attribute')).toBeUndefined();
  });*/
});
