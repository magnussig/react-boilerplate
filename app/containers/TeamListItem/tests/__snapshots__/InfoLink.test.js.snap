// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InfoLink /> should match the snapshot 1`] = `
.c0 {
  height: 100%;
  color: black;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
}

<a
  className="c0"
  href="/"
  onClick={[Function]}
/>
`;
